app/models/differential_expression_result.rb

```diff
@@ -13,22 +13,31 @@ class DifferentialExpressionResult
     DEFAULT_COMP_METHOD, 'logreg', 't-test', 't-test_overestim_var'
   ].freeze
 
+  # analysis types, e.g. A vs. B and A vs. All
+  DEFAULT_ANALYSIS = 'a-vs-all'.freeze
+  ANALYSIS_TYPES = [DEFAULT_ANALYSIS, 'a-vs-b'].freeze
```

Comment:
> Our [product docs](https://broadworkbench.atlassian.net/browse/SCP-4860) note the PM suggested just "asking for a single .csv file that contains the DE results for all comparisons", and our SME thought this would make sense.

Oh I totally misunderstood that - I had somehow conflated "comparisons" to "annotations".  Yes, in re-reading this it seems that the model would be for a single annotation per file.  I can make that change.

Reply:
> I suspect it'd be simpler implementationally to have users upload a single CSV per annotation that contains DE results for _any_ comparisons in that annotation -- whether it's one-vs-rest, pairwise, just one type, both, covering every group, some incomplete subset of groups, etc. Then, the process of [splitting the user's custom DE file](https://broadinstitute.slack.com/archives/CBEHTH601/p1680281337118449?thread_ts=1680279219.304729&cid=CBEHTH601) into per-comparison files for use in the DE UI could be more delegated to Ingest Pipeline.

I think a quick chat about this might be helpful - there's some detail that I'm failing to glean from re-reading Slack conversations.

Reply:
[922ab25](https://github.com/broadinstitute/single_cell_portal_core/pull/1753/commits/922ab25a49483d5e2c0aee0c5667fedf8a79a4d7)